test/e2e/instance-create.e2e.ts

```diff
@@ -229,3 +230,58 @@ test('shows object not found error on no default pool', async ({ page }) => {
+  await page.goto('/projects/mock-project/instances-new')
+  await page.getByRole('textbox', { name: 'Name', exact: true }).fill(instanceName)
+  await page.getByRole('tab', { name: 'Existing disks' }).click()
+  await page.getByRole('button', { name: 'Create instance' }).click()
+  await expect(page).toHaveURL(`/projects/mock-project/instances/${instanceName}/storage`)
+  await expectVisible(page, [`h1:has-text("${instanceName}")`, 'text=8 GiB'])
+  await expectRowVisible(page.getByRole('table'), { name: 'disk-3' })
+})
+
+test('create instance with a different existing disk', async ({ page }) => {
+  const instanceName = 'my-existing-disk-2'
+  await page.goto('/projects/mock-project/instances-new')
+  await page.getByRole('textbox', { name: 'Name', exact: true }).fill(instanceName)
+  await page.getByRole('tab', { name: 'Existing disks' }).click()
+  await page.getByLabel('Disk', { exact: true }).click()
+  await page.getByRole('option', { name: 'disk-4' }).click()
+  await page.getByRole('button', { name: 'Create instance' }).click()
+  await expect(page).toHaveURL(`/projects/mock-project/instances/${instanceName}/storage`)
+  await expectVisible(page, [`h1:has-text("${instanceName}")`, 'text=8 GiB'])
+  await expectRowVisible(page.getByRole('table'), { name: 'disk-4' })
+})
+
+test('start with an existing disk, but then switch to a silo image', async ({ page }) => {
+  const instanceName = 'silo-image-instance'
+  await page.goto('/projects/mock-project/instances-new')
+  await page.getByRole('textbox', { name: 'Name', exact: true }).fill(instanceName)
+  await page.getByRole('tab', { name: 'Existing disks' }).click()
+  await page.getByLabel('Disk', { exact: true }).click()
+  await page.getByRole('option', { name: 'disk-7' }).click()
+  await page.getByRole('tab', { name: 'Silo images' }).click()
+  await page.getByRole('button', { name: 'Create instance' }).click()
+  await expect(page).toHaveURL(`/projects/mock-project/instances/${instanceName}/storage`)
+  await expectVisible(page, [`h1:has-text("${instanceName}")`, 'text=8 GiB'])
+  await expectNotVisible(page, ['text=disk-7'])
+})
+
+test('maintains selected values even when changing tabs', async ({ page }) => {
+  await page.goto('/projects/mock-project/instances-new')
+  await page.getByRole('button', { name: 'Image' }).click()
+  // select the arch option
+  await page.getByRole('option', { name: 'arch-2022-06-01' }).click()
+  // expect to find name of the image on page
+  await expect(page.getByText('arch-2022-06-01')).toBeVisible()
+  // change to a different tab
+  await page.getByRole('tab', { name: 'Existing disks' }).click()
+  // the image should no longer be visible
+  await expect(page.getByText('arch-2022-06-01')).toBeHidden()
+  // change back to the tab with the image
+  await page.getByRole('tab', { name: 'Silo images' }).click()
+  // arch should still be selected
+  await expect(page.getByText('arch-2022-06-01')).toBeVisible()
+})
```

Comment:
These look great. Too bad there's not really a way to check after create that the instance got the right image.

Reply:
Looks like we can verify that a substring from the original image's ID was used as part of the new disk's name.